src/App.js

```diff
@@ -1,11 +1,199 @@
+		});
+
+		// grab total
+		// update local state to the total stored in database
+		get(totalRef).then((snapshot) => {
+			setTotal(snapshot.val().cartTotal);
+		});
+	}, []);
+
+	useEffect(() => {
+		if (total !== 0) {
+			// only update database if the local total isn't its on-mount value of 0
+			updateTotalInDb({ cartTotal: total });
+		}
+	}, [total]);
+
+	const updateTotalInDb = (totalObj) => {
+		const database = getDatabase(firebase);
+		const totalRef = ref(database, `/total/-N0yHl4A1PhHDEClucsy`);
+		return update(totalRef, totalObj);
+	};
+
+	const addListItems = (burgerObj) => {
+		// update total in shopping cart
+		const { subtotal } = burgerObj;
+		setTotal(total + subtotal);
+
+		// add burger object to database
+		const database = getDatabase(firebase);
+		const cartRef = ref(database, `/cart/`);
+
+		// add items to cart in database
+		push(cartRef, burgerObj);
+	};
+
+	const handleRemove = (burgerId, subtotal) => {
+		const database = getDatabase(firebase);
+		const cartRef = ref(database, `/cart/${burgerId}`);
+
+		// update total in shopping cart
+		setTotal(total - subtotal);
+
+		// remove from database
+		remove(cartRef);
+
+		// if its the last item in the cart, then clear the database total
+		if (burgers.length === 1) {
+			updateTotalInDb({ cartTotal: 0 });
+		}
+	};
+
+	const clearCartInDb = () => {
+		// reset database cart
+		const database = getDatabase(firebase);
+		const cartRef = ref(database, `/cart/`);
+
+		remove(cartRef);
+	};
+
+	const handleCheckout = (event) => {
+		event.preventDefault();
+
+		// confirmation message
+		alert("Thank you for your purchase!");
+
+		// reset local state
+		setBurgers([]);
+		setTotal(0);
+
+		// reset database total
+		updateTotalInDb({ cartTotal: 0 });
+
+		// reset cart in database
+		clearCartInDb();
+	};
+
+	const handleModalOpen = (burgerProduct) => {
+		// open the modal
+		const modalElem = document.querySelector("#modal");
+		// make modal visible
+		modalElem.style.display = "block";
+		// update burger name state
+		const { name } = burgerProduct;
+		setModalBurger(name);
+	};
+
+	return (
+		<div className="App">
+			<Modal modalBurger={modalBurger} />
+			<div className="bg-img-container wrapper">
+				{/* Modal displays burger customization options */}
+				<header>
+					<div className="header-container wrapper">
+						<div className="title-container">
+							<h1>Mel's Drive-In</h1>
+						</div>
+						<div className="header-inner-container">
+							<div className="img-container">
+								<img
+									src="https://images.unsplash.com/photo-1513918573039-35f42ae13895?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1074&q=80"
+									alt="an old 50s car in front of a diner"
+								/>
+							</div>
+							<ShoppingCart
+								burgers={burgers}
+								total={total}
+								handleRemove={handleRemove}
+								handleCheckout={handleCheckout}
+							/>
+						</div>
+					</div>
+				</header>
+				<main>
```

Comment:
Similar idea here - could you create a component for the main or parts of it to reduce the size of this.

Reply:
Yep, that's a good idea for readability.